frontend/src/components/core/Block/Block.tsx

```diff
@@ -143,10 +144,20 @@ class Block extends PureComponent<Props> {
     const BlockType = deltaBlock.expandable ? this.WithExpandableBlock : Block
     const optionalProps = deltaBlock.expandable ? deltaBlock.expandable : {}
     let style: any = { width }
-    if (deltaBlock.column) {
+    if (deltaBlock.column && deltaBlock.column.weight) {
+      // 500px will be the minimal viewport width used to determine the
+      // minimal fixed column width while accounting for column proportions.
+      // Randomly selected based on visual experimentation.
+      const minViewportForColumns = 500
+
+      // When working with columns, width is driven by what percentage of space
+      // the column takes in relation to the total number of columns
+      const columnPercentage = deltaBlock.column.weight / width
+      const minColumnWidth = columnPercentage * minViewportForColumns
       style = {
         // Flex determines how much space is allocated to this column.
         flex: deltaBlock.column.weight,
+        minWidth: `max(${columnPercentage * 100}% - 8px, ${minColumnWidth}px)`,
```

Comment:
If we end up enforcing a maximum number of columns, we could perhaps make this calculation smarter but I think  in order to do we probably need configurable padding